huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_batchtask_test.go

```diff
@@ -0,0 +1,295 @@
+	)
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&obj,
+		getBatchTaskResourceFunc,
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckIOTDABatchTaskFilePath(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testBatchTask_withTargetsFileField(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "type", "createDevices"),
+
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "status"),
+					resource.TestCheckResourceAttrSet(resourceName, "created_at"),
+					resource.TestCheckResourceAttrSet(resourceName, "task_details.#"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"space_id", "targets", "targets_filter", "targets_file"},
+			},
+		},
+	})
+}
+
+func TestAccBatchTask_derived(t *testing.T) {
+	var (
+		obj          interface{}
+		resourceName = "huaweicloud_iotda_batchtask.test_derived"
+		name         = acceptance.RandomAccResourceName()
+	)
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&obj,
+		getBatchTaskResourceFunc,
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckHWIOTDAAccessAddress(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testBatchTask_derived(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", name),
+					resource.TestCheckResourceAttr(resourceName, "type", "unfreezeDevices"),
+					resource.TestCheckResourceAttr(resourceName, "status", "Fail"),
+
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "status_desc"),
+					resource.TestCheckResourceAttrSet(resourceName, "created_at"),
+					resource.TestCheckResourceAttr(resourceName, "task_details.#", "0"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"space_id", "targets", "targets_filter", "targets_file"},
+			},
+		},
+	})
+}
+
+func testBatchTask_base(name, nodeId string) string {
+	productBasic := testProduct_basic(name)
+
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_iotda_device" "test1" {
+  name        = "%[2]s"
+  node_id     = "%[3]s"
+  space_id    = huaweicloud_iotda_space.test.id
+  product_id  = huaweicloud_iotda_product.test.id
```

Comment:
There is an extra space to the left of the equals sign.

Reply:
fixed.